src/java.base/share/classes/java/util/concurrent/ThreadPoolExecutor.java

```diff
@@ -1256,13 +1256,21 @@ public ThreadPoolExecutor(int corePoolSize,
                               BlockingQueue<Runnable> workQueue,
                               ThreadFactory threadFactory,
                               RejectedExecutionHandler handler) {
-        if (corePoolSize < 0 ||
-            maximumPoolSize <= 0 ||
-            maximumPoolSize < corePoolSize ||
-            keepAliveTime < 0)
-            throw new IllegalArgumentException();
-        if (workQueue == null || threadFactory == null || handler == null)
-            throw new NullPointerException();
+        if (corePoolSize < 0) {
+            throw new IllegalArgumentException("corePoolSize can't be negative, but got " + corePoolSize);
+        } else if (maximumPoolSize <= 0) {
+            throw new IllegalArgumentException("maximumPoolSize must be positive, but got " + maximumPoolSize);
+        } else if (maximumPoolSize < corePoolSize) {
+            throw new IllegalArgumentException("maximumPoolSize must >= corePoolSize , " + "but got maximumPoolSize:" + maximumPoolSize + " corePoolSize :" + corePoolSize);
```

Comment:
Too many different formulations of messages for me ("can't" vs "must be" vs "must >=")

I'd recommend using "must be" as in "must be non-negative", "must be positive", "must be greater-or-equal to"